contracts/schemes/TransitionScheme.sol

```diff
@@ -0,0 +1,88 @@
+pragma solidity 0.5.17;
+
+import "../controller/Controller.sol";
+
+/**
+ * @title A scheme for transitioning the DAO's assets to a new one.
+ */
+
+contract TransitionScheme {
+
+    Avatar public avatar;
+    address payable public newAvatar;
+    address[] public externalTokens;
+    address[] public assetAddresses;
+    bytes4[] public selectors;
+
+    /**
+     * @dev initialize
+     * @param _avatar the avatar to migrate from
+     * @param _newAvatar the avatar to migrate to
+     * @param _externalTokens external tokens to allow transfer to the new avatar
+     * @param _assetAddresses the assets to transfer
+     * @param _selectors the functions to call to to transfer the assets
+     */
+    function initialize(
+        Avatar _avatar,
+        address payable _newAvatar,
+        address[] calldata _externalTokens,
+        address[] calldata _assetAddresses,
+        bytes4[] calldata _selectors
+    ) external {
+        require(_assetAddresses.length == _selectors.length, "Arrays length mismatch");
+        require(avatar == Avatar(0), "can be called only one time");
+        require(_avatar != Avatar(0), "avatar cannot be zero");
+        avatar = _avatar;
+        newAvatar = _newAvatar;
+        externalTokens = _externalTokens;
+        assetAddresses = _assetAddresses;
+        selectors = _selectors;
+    }
+
+    /**
+     * @dev transferAssets function
+     * transfer the DAO assets to a new DAO
+     */
+    function transferAssets() external {
+        for (uint256 i=0; i < assetAddresses.length; i++) {
+            Controller(avatar.owner()).genericCall(
+                assetAddresses[i],
+                abi.encodeWithSelector(selectors[i], newAvatar),
```

Comment:
is that assume that the selector to transfer always get one param (address) ?

Reply:
Yes. I didn't want to allow any data, though it shouldn't be a security risk so long as it is properly verified when doing the initialize